sklearn/cluster/dbscan_.py

```diff
@@ -353,6 +362,7 @@ def fit(self, X, y=None, sample_weight=None):
 
         """
         X = check_array(X, accept_sparse='csr')
+        _validate_bad_defaults(self)
```

Comment:
should this be a method in `BaseEstimator` instead?
We're introducing other validation tools in the base class (e.g. `self._validate_X()`)

Reply:
I thought about that too (especially sine the only parameter to the function is `self`). But this way it would be less intrusive to the codebase. `validate_X` is something which is used in practically all estimators, but `_validate_bad_defaults` will ideally be rarely used.

I have no strong opinions on that, I can move it to the `BaseEstimator`